ios/RNShopify.m

```diff
@@ -0,0 +1,133 @@
+
+#import "RNShopify.h"
+#import "Buy.h"
+
+@implementation RNShopify {
+  RCTPromiseResolveBlock _resolve;
+  RCTPromiseRejectBlock _reject;
+}
+
+- (dispatch_queue_t)methodQueue
+{
+    return dispatch_get_main_queue();
+}
+RCT_EXPORT_MODULE()
+
+RCT_EXPORT_METHOD(initialize:(NSString *)domain key:(NSString *)key)
+{
+
+  self.client = [[BUYClient alloc] initWithShopDomain:domain
+                                                     apiKey:key
+                                                      appId:@"8"];
+}
+
+RCT_EXPORT_METHOD(getShop:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+
+  [self.client getShop:^(BUYShop *shop, NSError *error) {
+    if (shop && !error) {
+      resolve(shop.domain);
+    }
+    else {
+      reject(@"no_shop", @"There was no shop", error);
+    }
+  }]; 
+}
+
+RCT_EXPORT_METHOD(getProducts:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+
+   [self.client getProductsPage:1 completion:^(NSArray *products, NSUInteger page, BOOL reachedEnd, NSError *error) {
```

Comment:
There's no way to define the offset in the SDK, and also, the page size is always to 25. The argument to change this is deprecated and doesn't work. We need to manage this ourselves and only send the page number.